botocore/__init__.py

```diff
@@ -70,6 +67,11 @@ def xform_name(name, sep='_', _xform_cache=_xform_cache):
         return name
     key = (name, sep)
     if key not in _xform_cache:
+        if _special_case_transform.search(name) is not None:
+            is_special = _special_case_transform.search(name)
```

Comment:
Interesting. Makes sense. :+1: 

In this case, perhaps we would like to add an extra line of comment as something like "# The following two lines are indeed faster in common case". This comment suggestion is definitely optional though.